### ISSUE TITLE
Discord Bot Integration

### ISSUE THREAD

**Issue by llamositopia** [llamositopia/kitboga-codeville#11]
Is Discord Bot integration worth adding? For implementation, the source can be hosted here, and for hosting I can host and run the Discord Bot to send updates and enable commands in order to allow some basic Issue/PR queries/updates to be sent to the Discord channel if approved by the staff team.

Is this something people would be interested in having?

It enables more people to be exposed to the repository and see changes more frequently, although it also adds extra messages to the channel when an updates happens to the repository.

**Comment by yukihyou** [comment]
Updates can be added to Discord channels using a Webhook - This is native GitHub functionality.

https://gist.github.com/jagrosh/5b1761213e33fc5b54ec7f6379034a22

**Comment by llamositopia** [comment]
I was thinking some commands could also be added to generate and view links on the repository. Although if we use web hooks, is this still something desired?

**Comment by yukihyou** [comment]
The webhooks will contain links to the given issue/commit/whatever, and should serve as enough 'discovery engagement' IMHO. If we get to the point where we want to control the repo from Discord, then we can revisit.

That said, having a `!coderepo` command or something in Bogabot would be nice?

**Comment by TheDevFreak** [comment]
I would happily write a discord bot for this repo, since a github api exists.

**Comment by PixeLInc** [comment]
Yeah as yuki stated above, You should use the default github webhooks that are supported by Discord since they provide pushes/stars/prs/comments and more.. No need to rewrite that.

As for 'controlling the repo' via the bot, the only thing that may be necessary for is imo having an issue command or something that allows users to throw up an issue or suggestion via the bot but other than that idk.

If you do go with a webhook - that's not necessarily apart of the bot so we wouldn't need anything if we go with that other than to slap a link in the webhook settings on Discord.
But if you do want those commands and such, such as like `!source` or something for people to get the github link or other things then a little bot would be necessary.

**Comment by yukihyou** [comment]
> But if you do want those commands and such, such as like !source or something for people to get the github link or other things then a little bot would be necessary.

A simple text response to a command can probably be added to BogaBot. And, if we have commits/issues/etc being pushed to Discord via the webhook, there will be links in those embeds back to the site anyway (so it shouldn't really have to be linked otherwise).

As for creating issues/etc from Discord - personally I feel that's not necessary, as currently the only things that would get raised would require reviewing the current codebase anyway - and at that point you're already on the site. That said - writing a "log an issue bot" is something that could be generically useful, and would be a handy portfolio project for someone.
